test/refract/test-JsonSchema.cc

```diff
@@ -259,6 +309,23 @@ SCENARIO("JSON Schema serialization of StringElement", "[json-schema]")
             }
         }
     }
+
+    GIVEN("A named StringElement with content and fixed attribute true")
+    {
+        auto el = from_primitive("foo");
+        el->element("Flip");
+        el->attributes().set("fixed", from_primitive(true));
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches the string \"foo\"")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","definitions":{"Flip":{"type":"string","enum":["foo"]}},"$ref":"#/definitions/Flip"})");
```

Comment:
I think it depends how you read the standard: http://json-schema.org/latest/json-schema-core.html#rfc.section.8
```
An object schema with a "$ref" property MUST be interpreted as a "$ref" reference. The value of the "$ref" property MUST be a URI Reference. Resolved against the current URI base, it identifies the URI of a schema to use. All other properties in a "$ref" object MUST be ignored.
```
You might have a point. I don't think it's invalid JSON Schema, although it's quite possible a parser might ignore the `schema` property. The URI is still valid, the fact a JSON Schema parser should ignore properties coinciding with `$ref` doesn't change the fact. Also, I've tried all examples against at least one linter without issues and this idiom seems to be common place on discussion forums.

We could wrap the root in an `allOf` though, just to be on the safe side. What do you think?

Reply:
I was able to get the following schema to work correctly on various validator implementations I tested. So seems it could be allowed.

```json
{
  "$schema":"http://json-schema.org/draft-04/schema#",
  "definitions":{
    "Items":{"type":"array", "items": {"$ref":"#/definitions/Item"}},
    "Item": {"type": "object"}
  },
  "$ref":"#/definitions/Items"
}
```